walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -13,11 +13,117 @@
 // limitations under the License.
 
 package com.google.sps;
-
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    // throw new UnsupportedOperationException("TODO: Implement this method.");
+    HashSet<String> attendees = new HashSet<String>(request.getAttendees());
+    HashSet<String> optionalAttendees = new HashSet<String>(request.getOptionalAttendees());
+    ArrayList<TimeRange> badTimes = new ArrayList<TimeRange>();
+    ArrayList<TimeRange> optionalBad = new ArrayList<TimeRange>();
+    Collection<TimeRange> range = new ArrayList<TimeRange>();
+    Collection<TimeRange> OptRange = new ArrayList<TimeRange>();
```

Comment:
nice variable names here

Reply:
Also this is another picky thing that I'm not actually sure about: I've heard that it's a good idea to make your variables as general as possible so it would be easy to change their type later. I don't think this is necessary, but I usually try to say Set<String> attendees = new HashSet.... or List<TimeRange> badTimes = new ArrayList... The left side of the = I usually have as the general type, but I'm not sure if that's necessary. If Jiaxin or Guillermo could confirm about this that would be great! Sorry for very long comment :)

Reply:
That's a good idea, I will keep this in mind for the capstone project!

Reply:
Can confirm it's usually good to keep variables as generic as reasonable. This is particularly true for parameters into methods/constructors. Return types of methods however should be as specific as reasonable though. This is so that methods can accept as many input types as they can and return types can satisfy as many output types as possible (e.g. if I return an ArrayList<> it can be assigned to a Collection<> as used above.